js/maybe/Maybe.ts

```diff
@@ -86,20 +81,16 @@ class Maybe {
 
     const rejectedEntry = maybeArray.find((maybe) => maybe.isRejected());
     if (rejectedEntry) {
-      return rejectedEntry;
+      return rejectedEntry; // Returns Maybe<U> instead of Maybe<U[]> - cast to array @pierce?
```

Comment:
This causes an annoying type issue where `all` will return `Maybe<U[]>` except in this one specific case, where `Maybe<U>` is returned. I believe this is to have the first error to be available synchronously. Since we return a `Maybe`, the interface is the same for consumers, but consumers will always have to check if `Maybe<U>.value` is an array or not, which becomes more messed up when `U` is an array type, such as `string[]`, then forcing consumers to always check `Maybe<U>.isRejected` before continuing.

Reply:
`Maybe<never>` throws millions of errors.

If we split to `_value` and `_error` and update `_fromError` to set `_error` instead of value, we may be able to get rid of `Maybe<unknown>`...

Reply:
The one difference that is throwing off the types here is Promise.all `throw`s the first rejected thing instead of returning. If we throw `rejectedEntry` here, which is consistent with the `value` method, we should be able to simplify the return

Reply:
As long as we maintain the behavior (we return a maybe that is rejected for the same rejected value as the first rejected maybe in the array).